src/natcap/invest/utils.py

```diff
@@ -746,3 +746,27 @@ def _assert_vectors_equal(
         expected_vector = None
 
     return None
+
+
+def is_valid(array, nodata):
+    """
+    Return boolean array showing the valid elements of the array.
+
+    This is to be used for masking the nodata pixels of a raster. Assuming that
+    the input array is numeric and contains no None or numpy.nan elements.
+    If ``nodata=None``, that implies that all pixels are valid.
+
+    Args:
+        array (numpy.ndarray): Array for which to make boolean mask
+        nodata: Nodata value to compare to. 
+
+    Returns:
+        boolean numpy.ndarray with the same shape as the input array. 
+        Each element is False if its value is close to the nodata value,
+        True otherwise.
+
+    """
+    if nodata:
+        return ~numpy.isclose(array, nodata)
+    else:
+        return array != nodata
```

Comment:
There are a couple of things that will probably turn out to be problematic here:

1. Python's truthiness rules allow for values of `0`, `False`, empty iterables and a few other things to all be interpreted as `False`.  How will this function behave if we have a nodata value of `0`, which is a valid input?
2. similar to @richpsharp 's comment at https://github.com/natcap/invest/issues/228#issuecomment-675160937, the function name `is_valid` makes it sound like this function is for checking whether a pixel is valid ... but all it's actually checking for is whether the value is close to `nodata`.  Function names seem like a fairly trivial thing, but it's so so important for us to be clear about the intent of a function and what it's really trying to do.  So if the intent is to determine whether a value is close to `nodata`, then we should really be expressing that in the title.  What, exactly, is a valid pixel, is highly dependent on the use case.
3. If the intent here is actually to create a single function that checks to see if a pixel value is close to `nodata`, then we need to accommodate the differences in floating-point and integer matrices.  It turns out that if we know in advance that we have an integer matrix, then we can take advantage of order-of-magnitude speedups in per-element comparisons: `integer_array != nodata` is an order of magnitude faster than `numpy.isclose(integer_array, nodata)`.   On very large datasets, this can make a noticeable difference.  Here's the timing details:

```python
>>> timeit.timeit('numpy.isclose(ones, 0)', setup='import numpy; ones=numpy.ones((50, 50), dtype=numpy.float32)', number=100000)
5.014724600000022
>>> timeit.timeit('numpy.isclose(ones, 0)', setup='import numpy; ones=numpy.ones((50, 50), dtype=numpy.int16)', number=100000)
4.701722299999915
>>> timeit.timeit('ones != 0', setup='import numpy; ones=numpy.ones((50, 50), dtype=numpy.int16)', number=100000)
0.315173499999986
```
4. By making it necessary to have an extra function callthrough, we're not only obfuscating the intent of the function, there's also a (relatively minor) computational overhead.  The main point here is that the value gained from having a function should outweigh the cognitive and computational overhead of needing to look elsewhere at what the function is doing.  This function is so simple that I'm not sure this threshold is met.